sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/ExceptionExtensions.cs

```diff
@@ -48,5 +49,29 @@ internal static class ExceptionExtensions
                     return instance;
             }
         }
+
+        /// <summary>
+        ///   Considers an exception surfaced during the creation of an AMQP link, determining if the cause was a race condition
+        ///   with the connection closing and translating it into the form that should be considered for error handling decisions.
+        /// </summary>
+        ///
+        /// <param name="instance">The instance that this method was invoked on.</param>
+        /// <param name="eventHubName">The name of the Event Hub that the service operation was targeting.</param>
+        ///
+        /// <returns>The <see cref="Exception" /> that corresponds to the <paramref name="instance" /> and which represents the service error.</returns>
+        ///
+        public static Exception TranslateConnectionCloseDuringLinkCreationException(this InvalidOperationException instance,
+                                                                                    string eventHubName)
+        {
+            Argument.AssertNotNull(instance, nameof(instance));
+
+            return instance switch
+            {
+                _ when (CultureInfo.InvariantCulture.CompareInfo.IndexOf(instance.Message, "when the connection is closing", CompareOptions.IgnoreCase) >= 0) =>
```

Comment:
I loathe this, but after looking through the code in the AMQP library, I can't see a better way to target this without assuming all `InvalidOperationExceptions` in the given scope have this root cause.